src/documents/IoCTestingFramework.md

```diff
@@ -122,151 +116,398 @@ fluid.tests.globalCatTest = function (catt) {
+            <td><code>{String}</code></td>
+            <td>If present, will override the key of this member of <code>elements</code> in representing the namespace of this element.
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+The members `gradeNames` and `options` in the above table collectively designate an Infusion component derived from `fluid.test.sequenceElement` &mdash; this component will be 
+instantiated as a child component of the `fluid.test.testCaseHolder` and its options member `sequence` will be evaluated. This member `sequence` holds one or more 
+[`fixture elements`](#supported-fixture-records) just as seen in the top-level [`sequence`](#-modules-tests-and-sequence-) element.
+
+Any fixture elements listed at the traditional top-level `sequence` member will be grandfathered in to this system with a namespace of `sequence` &mdash; for example, a `testSequenceElement` with
+a priority of `"before:sequence"` will be sorted before these elements.
+
+#### Example of sequence building using `sequenceGrade`
+
+Here is a simple example of building up a sequence of elements piece by piece, using [priority](Priorities.md) constraints of the type `after:<namespace>` and `before:<namespace>`. At the top
+level, we define a compound `testEnvironment` and `testCaseHolder` that defines a single module holding a single test, referencing the grade `fluid.tests.elementPrioritySequence` as 
+designating the overall sequence to be built up as a `sequenceGrade`. This grade is defined with four `testSequenceElement` entries, whose namespaces are `check`, `end`, `postBeginning` and `beginning`
+(deliberately defined in a different order to that which they will be executed in). These elements each specify a `priority` element to guide the order in which they should be sorted into a sequence.
+If you run this example, you will see them executed in the order `beginning`, `postBeginning`, `sequence` (the grandfathered-in top-level sequence), `end` and finally `check`.
+
+The sequence makes use of two reusable `sequenceElement` grades, `fluid.tests.elementPriority.log` which defines a sequence element which logs a console message (this fixture element uses
+the [compact syntax](Invokers.md#compact-format-for-invokers) for encoding simple argument lists together with the function), and `fluid.tests.elementPriority.check` which defines
+a sequence element making a jqUnit assertion which always passes.
+
+```javascript
+fluid.defaults("fluid.tests.elementPriority.log", {
+    gradeNames: "fluid.test.sequenceElement",
+    sequence: [{
+        func: "fluid.log({that}.options.message)"
+    }]
+});
+
+fluid.defaults("fluid.tests.elementPriority.check", {
+    gradeNames: "fluid.test.sequenceElement",
+    sequence: [{
+        func: "jqUnit.assert",
+        args: "I am the check, right at the end"
+    }]
+});
+
+fluid.defaults("fluid.tests.elementPrioritySequence", {
+    gradeNames: "fluid.test.sequence",
+    sequenceElements: {
```

Comment:
@the-t-in-rtf - I think that you or your package manager may have made a mistake in updating to Infusion 2.0. I tested the sample in your code snippet derived from that in the docs and it gives the expected results. However I filed the "obscure diagnostic" issue as https://issues.fluidproject.org/browse/FLUID-6097